### PR TITLE
Add mTLS support to Mixer MCP stack

### PR DESCRIPTION
**Pull request by ozevren**
Add mTLS support to Mixer MCP stack.

- Add a command-line flag to specify the location of the certificate files.
- Jam the parameter through the config initialization stack to the MCP backend (sorry, I am not proud of this).
- Optionally enable using secure mode (based on the URL)

### CODE REVIEW

File: galley/pkg/server/args.go [hunk 1]
```diff
@@ -24,6 +24,15 @@ import (
 	"istio.io/istio/pkg/probe"
 )
 
+const (
+	defaultCertDir         = "/etc/istio/certs/"
+	defaultConfigMapFolder = "/etc/istio/config"
```
Comment (costinm) [inline galley/pkg/server/args.go]: New Istio configs should go under /var/lib/istio

File: pkg/mcp/creds/create.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
+//  Copyright 2018 Istio Authors
```
Comment (costinm) [inline pkg/mcp/creds/create.go]: Please move watcher and this file under /pkg/mtls ( or similar name ) - ads client and server will need to use it as well ( we don't have rotation yet in pilot - that's why we still use envoy in front )

File: galley/cmd/galley/cmd/server.go [hunk 3]
```diff
@@ -46,7 +46,16 @@ func serverCmd(printf, fatalf shared.FormatFn) *cobra.Command {
...
-
+	cmd.PersistentFlags().BoolVarP(&sa.Insecure, "insecure", "", sa.Insecure,
+		"Use insecure gRPC communication")
+	cmd.PersistentFlags().StringVarP(&sa.CertificateFile, "certFile", "", sa.CertificateFile,
+		"The location of the certificate file for mutual TLS")
+	cmd.PersistentFlags().StringVarP(&sa.KeyFile, "keyFile", "", sa.KeyFile,
+		"The location of the key file for mutual TLS")
+	cmd.PersistentFlags().StringVarP(&sa.CACertificateFile, "caCertFile", "", sa.CACertificateFile,
+		"The location of the certificate file for the root certificate authority")
+	cmd.PersistentFlags().StringVarP(&sa.ConfigMapFolder, "configMapFolder", "", sa.ConfigMapFolder,
```
[the remark this answers is not in the document]
Comment (ayj) [reply galley/cmd/galley/cmd/server.go]: Ah, then this should be the full path to the configuration file (via configmap). I think I made the comment elsewhere that we shouldn't hardcode the accesscontrol list.

**Comment by ozevren** [comment]
@costinm  I think you reviewed this PR and left a comment saying moving the mtls code out to a top-level package. That was part of a previous checkin that this pack also contained/depended on. The original code is in, bur either me or Jason can do the change to move the certificiate/watcher code to a top-level package. (@ayj).

Otherwise all code review comments should be addressed.

**Pull request closed without merging**